src/Stryker.Core/Stryker.Core/Helpers/RoslynHelper.cs

```diff
@@ -198,13 +198,28 @@ public static bool ContainsNodeThatVerifies(this SyntaxNode node, Func<SyntaxNod
                        && (child.Parent is not LocalFunctionStatementSyntax localFunction || localFunction.ExpressionBody != child);
             } ).Any(predicate);
 
-        public static bool HasAMemberBindingExpression(this ExpressionSyntax expression) =>
-            expression switch
+        /// <summary>
+        /// Find the syntax node or token that is just before the given node.
+        /// </summary>
+        /// <param name="node"></param>
+        /// <returns>previous node or token. Null if non exists.</returns>
+        public static SyntaxNodeOrToken GetPreviousSyntaxNode(this SyntaxNode node)
+        {
+            SyntaxNodeOrToken previous = node.Parent;
+            if (previous == null)
             {
-                MemberBindingExpressionSyntax => true,
-                MemberAccessExpressionSyntax memberAccess => memberAccess.Expression.HasAMemberBindingExpression(),
-                InvocationExpressionSyntax invocation => invocation.Expression.HasAMemberBindingExpression(),
-                _ => false
-            };
+                return null;
+            }
+
+            foreach(var child in previous.ChildNodesAndTokens())
+            {
+                if (child == node)
+                {
+                    return previous;
+                }
+                previous = child;
+            }
+            return null;
+        }
```

Comment:
When I wrote the original issue, I had imagined that something like `node.GetFirstToken(includeZeroWidth: true).GetPreviousToken(includeZeroWidth: true)` would suffice for a method like this. What led you to go with this logic instead?

Reply:
@dupdob I saw that you incorporated the approach I mentioned above. I assume it works as expected then? I came up with that approach after reaching some of Roslyn's documentation and source code, but I haven't had the chance to test it myself yet, so I just wanted to make sure you weren't simply taking my word for it.

Reply:
It looks like it gives the same result. I tried because it allowed me to remove the `FindPreviousNode` implementation.
I glanced at the implementation of `GetPreviousToken` and it looks overkill and a bit inefficient for this use case, but it should be fast enough.